File: Data Structures & Algorithms/majority-element/submission-1.java
class Solution {
    public int majorityElement(int[] nums) {
        int cnt = 0,ans = -1;
        for(int i = 0; i < nums.length; i++)
        {
            if(cnt == 0) ans = nums[i];
            if(nums[i] == ans) cnt+=1;
            else cnt-=1;
        }
        return ans;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int>pr(n),sf(n);
        int p = 1;
        for(int i = 0; i < n; i++)
        {
            pr[i] = p;
            p*=nums[i];
        }
        p = 1;
        for(int i = n - 1; i >= 0; i--)
        {
            sf[i] = p;
            p*=nums[i];
        }
        vector<int>ans(n);
        for(int i = 0; i < n; i++) ans[i] = pr[i]*sf[i];
        return ans;
    }
};